### PR TITLE
reqlog Fix garbage on stdout.

### PR DESCRIPTION
**Pull request by jmleddy**
The problem was that instead of dump I had used logmsg before.

### CODE REVIEW

File: db/reqlog.c [hunk 1]
```diff
@@ -1227,16 +1227,24 @@ int reqlog_logf(struct reqlogger *logger, unsigned event_flag, const char *fmt,
...
+        if (logger->event_mask & event_flag) {
+            struct print_event *event;
+            event = malloc(sizeof(struct print_event));
+            if (!event) {
+                logmsg(LOGMSG_ERROR, "%s:malloc failed\n", __func__);
+                return -1;
+            }
+            event->event_flag = event_flag;
+            event->length = -1; /* to indicate length is unknown */
+            strcpy(event->text, s);
```
Comment (adizaimi) [inline db/reqlog.c]: Text is const char *text so you can't strcpy to it. Maybe strdup.

Comment (jmleddy) [inline db/reqlog.c]: Yes good point.

**Comment by adizaimi** [comment]
looks good. 
I think the way to test this one is:
'exec procedure sys.cmd.send("logmsg level debug")'
'exec procedure sys.cmd.send("reql results")'
'exec procedure sys.cmd.send("reql go")'
'select * from t1 limit 1'
and watch output in the db log should look something like:
2017/06/08 19:47:04 sql request 2 msec from ip-1-70-17-7 rc 0
2017/06/08 19:47:04   2 __memp_pg calls took 0 ms70-17-7 rc 0
2017/06/08 19:47:04   select * from t1 limit 1, ncols=2, rowcount=1, cost=20.000000
2017/06/08 19:47:04 row 1 = (1, x'1234')imit 1, ncols=2, rowcount=1, cost=20.000000
2017/06/08 19:47:04 ----------, x'1234')imit 1, ncols=2, rowcount=1, cost=20.000000

**Comment by riverszhang89** [comment]
```
(gdb) bt
#0  __strcpy_sse2_unaligned () at ../sysdeps/x86_64/multiarch/strcpy-sse2-unaligned.S:636
#1  0x00000000004ff85a in reqlog_logl (logger=0x6363c50, event_flag=1, s=0x8fe02a "blockop") at db/reqlog.c:1243
#2  0x0000000000428aff in handle_ireq (iq=0x7f49657b6048) at db/sltdbt.c:324
#3  0x00000000004259df in thd_req (vthd=0x61c5470) at db/handle_buf.c:513
#4  0x000000000089f183 in thr_func (arg=0x608ab58) at bb/comdb2_pthread_create.c:231
#5  0x00007f4972e00064 in start_thread (arg=0x7f49657b2700) at pthread_create.c:309
#6  0x00007f4971cec62d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
(gdb) f 1
(gdb) p event->text
$3 = 0x0
```

File: db/reqlog.c (hunk 1 above)
Comment (riverszhang89) [inline db/reqlog.c]: @adizaimi https://github.com/bloomberg/comdb2/blob/master/db/reqlog_int.h#L43
text is not const.

Comment (adizaimi) [inline db/reqlog.c]: Hmm, it was in one of the versions of that file which I saw. Sorry for the confusion, I don't know when it moved to the .h file.

**Comment by adizaimi** [comment]
Check for s at func start?

**Pull request merged**